ark/utils/load_utils.py

```diff
@@ -297,37 +315,43 @@ def load_imgs_from_dir(data_dir, imgdim_name='compartments', image_name='img_dat
                           f"because the loaded images are floats")
             dtype = data_dtype
 
-    if variable_sizes:
-        img_data = np.zeros((len(imgs), 1024, 1024, 1), dtype=dtype)
-    else:
-        img_data = np.zeros((len(imgs), test_img.shape[0], test_img.shape[1], 1),
-                            dtype=dtype)
+    # extract data
+    img_data = []
+    for img in imgs:
+        v = io.imread(os.path.join(data_dir, img))
```

Comment:
We should do these checks based on the `test_img` above, rather than each time we load a new image. If there are images of different sizes, I think there's a 99% chance that was a mistake, rather than someone intentionally loading data of different dimensions into the same xarray

Reply:
I agree that it is unlikely to intentionally load data that contains both standard and multi-tiffs, or different numbers of rows/columns. However I'm not sure why should we treat input that contains both channels_first and channels_last as a ValueError, as we can very easily handle it, and it doesn't change the implementation that much (perhaps the images came from different sources, so their format is different).

Reply:
changed